extensions/azure-functions-resteasy/runtime/src/main/java/io/quarkus/azure/functions/resteasy/runtime/Function.java

```diff
@@ -0,0 +1,76 @@
+package io.quarkus.azure.functions.resteasy.runtime;
+
+import java.util.Optional;
+
+import javax.ws.rs.ext.RuntimeDelegate;
+
+import org.jboss.resteasy.core.SynchronousDispatcher;
+import org.jboss.resteasy.mock.MockHttpResponse;
+import org.jboss.resteasy.specimpl.ResteasyUriInfo;
+
+import com.microsoft.azure.functions.ExecutionContext;
+import com.microsoft.azure.functions.HttpRequestMessage;
+import com.microsoft.azure.functions.HttpResponseMessage;
+import com.microsoft.azure.functions.HttpStatus;
+import com.microsoft.azure.functions.annotation.HttpTrigger;
+
+import io.quarkus.runtime.Application;
+
+public class Function {
+
+    public HttpResponseMessage run(
+            @HttpTrigger(name = "req") HttpRequestMessage<Optional<byte[]>> request,
+            final ExecutionContext context) {
+        if (request.getUri().getPath().endsWith("quarkus-status")) {
+            HttpResponseMessage.Builder responseBuilder = request
+                    .createResponseBuilder(HttpStatus.valueOf(200)).body(deploymentStatus.getBytes());
+            return responseBuilder.build();
+        }
+        return resteasyDispatch(request);
+    }
+
+    static String deploymentStatus = "ok";
+
+    static {
+        String[] args = {};
+        try {
+            Class appClass = Class.forName("io.quarkus.runner.ApplicationImpl1");
```

Comment:
Is this intended to work on substrate, or just the Azure JVM runtime (I am not even sure if native is supported on Azure). This would not work with native as it will start threads and open sockets as part of the startup process.

Reply:
Azure JVM only.   

There was no native option documented for Azure Functions.